lib/ex_aws/auth/presigned_posts/policy.ex

```diff
@@ -0,0 +1,34 @@
+defmodule ExAws.Auth.PresignedPosts.Policy do
+  @moduledoc false
+  import ExAws.Auth.Utils
+
+  defstruct [expiration: nil, conditions: []]
+
+  def new do
+    %__MODULE__{}
+  end
+
+  def set_expiration(policy, datetime) do
+    %__MODULE__{policy | expiration: datetime}
+  end
+
+  def add_condition(policy = %__MODULE__{conditions: conditions}, condition) do
+    new_conditions =
+      conditions
+      |> Enum.reverse
+      |> List.insert_at(0, condition)
+      |> Enum.reverse
+    %__MODULE__{policy | conditions: new_conditions}
```

Comment:
The policy submitted to amazon is an array, and I assume if you had 2 conflicting rules, they have logic about which one would win. I'd like to keep it so when you add rules to a policy, they come out in the same order you added them. However I do see this is not very efficient and maybe the Erlang `:queue` module would help.